lib/auth/github.go

```diff
@@ -952,7 +953,26 @@ func ValidateClientRedirect(clientRedirect string, ssoTestFlow bool, settings *t
 	case "http":
 		switch u.Hostname() {
 		default:
-			return trace.BadParameter("invalid hostname in client redirect URL")
+			if settings == nil || len(settings.InsecureAllowedCidrRanges) == 0 {
+				return trace.BadParameter("invalid hostname in client redirect URL")
+			}
+			hostIP, err := netip.ParseAddr(u.Hostname())
+			if err != nil {
+				return trace.BadParameter("custom client HTTP redirect URLs must be IP addresses")
+			}
+
+			for _, cidrStr := range settings.InsecureAllowedCidrRanges {
+				cidr, err := netip.ParsePrefix(cidrStr)
+				if err != nil {
+					return trace.Wrap(err)
+				}
```

Comment:
Since I check that the CIDR prefixes aren't malformed when the parent OIDC connector is created I think returning the error makes sense, as it should be a rare occurrence that's only due to an internal bug

Reply:
Fair, changed to log instead of returning the error